libraries/engage/product/components/RelationsSlider/RelationsSheet.connector.js

```diff
@@ -0,0 +1,17 @@
+import { connect } from 'react-redux';
+import { getMaximumRelatedProducts } from '@shopgate/pwa-common-commerce/product';
+
+/**
+ * @param {Object} state The application state.
+ * @param {Object} props The component props.
+ * @returns {Object}
+ */
+const mapStateToProps = (state, { productId, type, limit = 100 }) => ({
+  products: getMaximumRelatedProducts({
```

Comment:
`getMaximumRelatedProducts` is a selector factory. When you use it like that, a new selector instance is created whenever `mapStateToProps` invoked - so on every state update.

Reply:
Changed